File: src/main/java/com/github/mucahitkayadan/envmasker/file/EnvFileType.java
package com.github.mucahitkayadan.envmasker.file;

import com.intellij.openapi.fileTypes.FileType;
import com.intellij.openapi.util.IconLoader;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class EnvFileType implements FileType {
    public static final EnvFileType INSTANCE = new EnvFileType();

    @NotNull
    @Override
    public String getName() {
        return "Env file";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Environment variables file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "env";
    }

    @Override
    public Icon getIcon() {
        return IconLoader.getIcon("/icons/env.svg", getClass());
    }

    @Override
    public boolean isBinary() {
        return false;
    }

}

File: src/main/kotlin/org/example/envmasker/file/EnvFileType.kt
package org.example.envmasker.file

import com.intellij.openapi.fileTypes.FileType
import com.intellij.openapi.util.IconLoader
import javax.swing.Icon

object EnvFileType : FileType {
    override fun getName() = "Env file"
    override fun getDescription() = "Environment variables file"
    override fun getDefaultExtension() = "env"
    override fun getIcon(): Icon = IconLoader.getIcon("/icons/env.svg", javaClass)
    override fun isBinary() = false
    override fun isReadOnly() = false
}
